File: Daily Test/Java/Form Matrix - Rows & Columns Positions 15.9.2025.java
import java.util.*;

class Cell {
    int row, col;
    String val;
    Cell(int r, int c, String v) { row = r; col = c; val = v; }
}

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine();
        String[] parts = line.split("\\s+");

        List<Cell> list = new ArrayList<>();
        for (String token : parts) {
            StringBuilder nums = new StringBuilder();
            StringBuilder text = new StringBuilder();
            List<Integer> numbers = new ArrayList<>();

            for (char ch : token.toCharArray()) {
                if (Character.isDigit(ch)) nums.append(ch);
                else {
                    if (nums.length() > 0) {
                        numbers.add(Integer.parseInt(nums.toString()));
                        nums.setLength(0);
                    }
                    text.append(ch);
                }
            }
            if (nums.length() > 0) numbers.add(Integer.parseInt(nums.toString()));
            list.add(new Cell(numbers.get(0), numbers.get(1), text.toString()));
        }

        list.sort((a, b) -> a.row == b.row ? a.col - b.col : a.row - b.row);

        int currentRow = list.get(0).row;
        for (Cell cell : list) {
            if (cell.row != currentRow) {
                System.out.println();
                currentRow = cell.row;
            }
            System.out.print(cell.val + " ");
        }
    }
}

File: Daily Test/C++/Form Matrix - Rows & Columns Positions 15.9.2025.cpp
#include <bits/stdc++.h>
using namespace std;

struct Cell {
    int row, col;
    string val;
};

int main() {
    string line;
    getline(cin, line);
    stringstream ss(line);

    vector<Cell> cells;
    string token;
    while (ss >> token) {
        int r = 0, c = 0;
        string text;
        string num;
        for (char ch : token) {
            if (isdigit(ch)) num += ch;
            else {
                if (!num.empty()) {
                    if (r == 0) r = stoi(num);
                    else c = stoi(num);
                    num.clear();
                }
                text += ch;
            }
        }
        cells.push_back({r, c, text});
    }

    sort(cells.begin(), cells.end(),
         [](const Cell &a, const Cell &b) {
             return (a.row == b.row) ? a.col < b.col : a.row < b.row;
         });

    int currentRow = cells[0].row;
    for (auto &cell : cells) {
        if (cell.row != currentRow) {
            cout << "\n";
            currentRow = cell.row;
        }
        cout << cell.val << " ";
    }
}